### PR TITLE
Fix #3384 : Arrange strings file

### LINKED ISSUE #3384: Arrange string resources based on screens
**Describe the bug**

Currently string resources are not organised correctly and are scattered in `strings.xml` file.

This should be fixed by following these points:
1. Make sure that each string is getting used in one specific view only. For example `Home` is part of multiple views like `Navigation Drawer`, `Home-Toolbar`. So for this we should have separate strings for both.
2. All strings related to one activity/fragment should be kept together by mentioning the activity name on top of the strings in a comment. Example: `<!-- ProfileResetPinActivity -->`
3. All activities/fragment sections should be in correct alphabetical order based on activity/fragment name. (The actual strings name are not required to be in alphabetical order but the comments should be in alphabetical order). Example:
```
<!-- HomeActivity -->
<string name="random_string_related_to_home">ABC</string>
<!-- ProfileResetPinActivity -->
<string name="random_string_related_to_profile_resent_pin">ABC</string>
<string name="random_string_related_to_profile_resent_pin_2">XYZ</string>
```

Arrange strings file as per below criteria:

**Screen/Toolbar Titles**
Used in manifest label, toolbar title and activity label test.
Pattern: `activity_name_activity_title`

Example: `administrator_controls_activty_title`

**App Content Strings**
Pattern: `activity_name_view_component_type`

`administrator_controls_activity_profile_management_label`
`administrator_controls_activity_account_actions_label`
`administrator_controls_activity_logout_item_text`
`administrator_controls_activity_app_version_item`

**Shared Strings**
Pattern: `view_component_description`

reading_text_size_item_text_small
reading_text_size_item_medium

**Test Strings**
Pattern: `test_file_name_view_component_type`

Dialogs
Pattern: `dialog_name_view_component_type`

Grouping Pattern: 

  `<!-- AdministratorControlActivity -->`
….
….
  `<!-- GeneralAvailabilityUpgradeNoticeDialogFragment -->`
…..
   `<!-- ProgressDatabaseFullDialog -->`

### PR DESCRIPTION
**Pull request by MohitGupta121**
## Explanation

Fix #3384 : Arrange strings file as per below criteria:

**Screen/Toolbar Titles**
Used in manifest label, toolbar title and activity label test.
Pattern: `activity_name_activity_title`

Example: `administrator_controls_activty_title`

**App Content Strings**
Pattern: `activity_name_view_component_type`

`administrator_controls_activity_profile_management_label`
`administrator_controls_activity_account_actions_label`
`administrator_controls_activity_logout_item_text`
`administrator_controls_activity_app_version_item`

**Shared Strings**
Pattern: `view_component_description`

reading_text_size_item_text_small
reading_text_size_item_medium

**Test Strings**
Pattern: `test_file_name_view_component_type`

Dialogs
Pattern: `dialog_name_view_component_type`

Grouping Pattern: 

  `<!-- AdministratorControlActivity -->`
….
….
  `<!-- GeneralAvailabilityUpgradeNoticeDialogFragment -->`
…..
   `<!-- ProgressDatabaseFullDialog -->`

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: app/src/main/res/values/strings.xml
Comment (MohitGupta121) [inline app/src/main/res/values/strings.xml:1]: @adhiamboperes PTAL, please pass your 1st review on this PR, Thanks!

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,167 +1,151 @@
...
+  <string name="administrator_controls_activity_learner_analytics_text">Learner Study Analytics</string>
+  <!-- AdministratorControlsFragmentTestActivity -->
+  <string name="administrator_controls_fragment_test_activity_title">Administrator Controls Fragment Test Activity</string>
+  <!-- AdminAuthActivity -->
+  <string name="admin_auth_activity_title">Administrator Controls</string>
+  <string name="admin_auth_activity_add_profiles_title">Authorise to add profiles</string>
+  <string name="admin_auth_activity_access_controls_title">Authorise to access Administrator Controls</string>
+  <string name="admin_auth_activity_heading_text">Administrator Authorization Required</string>
+  <string name="admin_auth_activity_subject_text">Enter the Administrator PIN in order to create a new account.</string>
+  <string name="admin_auth_activity_controls_subject_text">Enter the Administrator PIN in order to access Administrator Controls.</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]: Why the word _subject_ here? Could we use subheading instead?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,167 +1,151 @@
...
+  <string name="admin_pin_activity_warning_text">Do <b>not</b> use a PIN you have set for personal accounts like banking or social security.</string>
+  <string name="admin_pin_activity_input_new_pin_hint">New 5-Digit PIN</string>
+  <string name="admin_pin_activity_input_confirm_pin_hint">Confirm 5-Digit PIN</string>
+  <string name="admin_pin_activity_pin_length_error">Your PIN should be 5 digits long.</string>
+  <string name="admin_pin_activity_wrong_pin_text">Please make sure that both PINs match.</string>
+  <string name="admin_pin_activity_submit_button_text">Save</string>
+  <!-- AdministratorControlsFragment -->
+  <string name="administrator_controls_fragment_general_label">General</string>
+  <string name="administrator_controls_fragment_edit_account_text">Edit account</string>
+  <string name="administrator_controls_fragment_profile_management_text">Profile Management</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]: I think this was meant to be `_label`, based on 
```suggestion
  <string name="administrator_controls_fragment_profile_management_label">Profile Management</string>
```
Screenshot_1700660844

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,167 +1,151 @@
...
+  <string name="administrator_controls_fragment_edit_profiles_text">Edit profiles</string>
+  <string name="administrator_controls_fragment_download_permissions_text">Download Permissions</string>
+  <string name="administrator_controls_fragment_topic_update_on_wifi_text">Download and update only on Wi-fi</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]: Please confirm that these are named **label** or **text** based on their correct usages.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,167 +1,151 @@
...
+  <string name="administrator_controls_fragment_topic_update_on_wifi_description">Topics will be downloaded and updated only on Wi-fi. Any downloads or updates of cellular data will be queued.</string>
+  <string name="administrator_controls_fragment_auto_update_topic_text">Automatically update topics</string>
+  <string name="administrator_controls_fragment_auto_update_topic_description">Downloaded topics that have new content available will automatically update.</string>
+  <string name="administrator_controls_fragment_app_information_text">App Information</string>
+  <string name="administrator_controls_fragment_app_version_text">App Version</string>
+  <string name="administrator_controls_fragment_account_actions_text">Account Actions</string>
+  <string name="administrator_controls_fragment_log_out_text">Log Out</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]: This section looks like it should be moved up to be below the `AdministratorControlActivity` section.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,167 +1,151 @@
...
+  <!-- AddProfileActivity -->
+  <string name="add_profile_activity_text">Add Profile</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]: I think this is the activity title, please rename accroding to the expected pattern

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1,167 +1,151 @@
...
+  <string name="add_profile_activity_wrong_confirm_pin_error_text">Please make sure that both PINs match.</string>
+  <string name="add_profile_activity_info_image_content_description">More information on 3-digit PINs.</string>
+  <string name="add_profile_activity_edit_image_content_description">Current profile picture</string>
+  <string name="add_profile_activity_edit_user_image_content_description">Edit profile picture</string>
+  <string name="add_profile_activity_pin_check_box_text">Create a 3-Digit PIN</string>
+  <string name="add_profile_activity_user_name_helper_text">Required</string>
+  <!-- AdminSettingsDialogFragment -->
+  <string name="admin_settings_dialog_input_pin_hint">Administrator\'s PIN</string>
+  <string name="admin_settings_dialog_heading">Access to Administrator Settings</string>
+  <string name="admin_settings_dialog_subject_text">Administrator\'s PIN required to change user\'s PIN</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]:
```suggestion
  <string name="admin_settings_dialog_message_text">Administrator\'s PIN required to change user\'s PIN</string>
```

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -1,167 +1,151 @@
...
+  <!-- CellularAudioDialogFragment -->
+  <string name="cellular_audio_dialog_okay_button_text">Ok</string>
+  <string name="cellular_audio_dialog_cancel_button_text">Cancel</string>
+  <string name="cellular_audio_dialog_title">Currently on Cellular Data</string>
+  <string name="cellular_audio_dialog_description">Streaming audio may use a lot of cellular data.</string>
+  <string name="cellular_audio_dialog_checkboxes_text">Don\'t show this message again</string>
+  <!-- ConceptCardFragment -->
+  <string name="concept_card_fragment_title">Concept Card</string>
+  <string name="concept_card_test_activity_one_button_text">Concept Card 1</string>
+  <string name="concept_card_test_activity_two_button_text">Concept Card 2</string>
```
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]:
```suggestion
  <string name="concept_card_test_activity_button_one_text">Concept Card 1</string>
  <string name="concept_card_test_activity_button_two_text">Concept Card 2</string>
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @MohitGupta121! I took a first pass to verify the general patterns that we had agreed on. There are some small changes needed.

Additionally, can we move **GeneralStrings** and **SharedStrings** sections to the bottom of the strings files, so that the screen based sections come first?

Lastly, please update the PR description as discussed.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: `administrator_controls_fragment_edit_profiles_text` & `administrator_controls_fragment_topic_update_on_wifi_text` are **Text** but as per your above screenshot `administrator_controls_fragment_download_permissions_text` is **Label**

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

**Comment by MohitGupta121** [comment]
Deleting this PR because of unnecessary commit history.

**Pull request closed without merging**